src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -766,6 +774,10 @@ namespace AppInstaller::CLI::Workflow
                 context.Reporter.Info() << availableUpgradesCount << ' ' << Resource::String::AvailableUpgrades << std::endl;

```

Comment:
I see that you are using different strings to report the count depending on whether there is a single package or multiple, but here we were using the same string for both cases. I think we should be doing the same for both messages; although I'm not sure which one would be better. I would lean towards using a single string regardless of the number of packages as I don't know if there are any languages that make any distinction beyond 1/many or that don't make distinctions at all.

Reply:
I looked a bit into pluralization and it's a whole rabbit hole one can go into. I found it interesting to look at the [rules](https://unicode-org.github.io/cldr-staging/charts/latest/supplemental/language_plural_rules.html) for all languages...
I found some internal references to string formats for putting all variants in a single string and using libraries for handling that. @JohnMcPMS does the current setup support something like that? @JohnMcPMS , @denelon , is this something we would want to handle?

Reply:
I saw that too, but figured it was a separate PR... thanks for pointing it out :D

Reply:
If we wanted to do something to handle plurals programmatically, then we could probably do it at the same time as the dynamic string replacement mentioned above.  I don't see it as important as the dynamic strings though, so depending on the complexity we might just leave a nice place for it.  I don't know if the english standard of making it "optional" like:
```
N installed package version(s) cannot be determined.
```
applies to the other supported languages.  But it might be an option.

Reply:
Should I go ahead and use a optional plural for now? `SourceOpenFailedSuggestion` uses it ("Failed when opening source(s); try the 'source reset' command if the problem persists.")...

Reply:
That sounds good to me.